src/components/Card.js

```diff
@@ -1,50 +1,83 @@
-class Card {
-  constructor(data, cardSelector, _handleImageClick) {
-    this._name = data.name;
-    this._link = data.link;
+import Api from "./Api";
+
+export default class Card {
+  constructor(
+    { name, link, _id, isLiked },
+    cardSelector,
+    handleCardClick,
+    handleTrashClick,
+    handleLikeClick
+  ) {
+    this._name = name;
+    this._link = link;
     this._cardSelector = cardSelector;
-    this._handleImageClick = _handleImageClick;
+    this._handleCardClick = handleCardClick;
+    this._handleTrashClick = handleTrashClick;
+    this._handleLikeClick = handleLikeClick;
+    this._cardId = _id;
+    this._isLiked = isLiked;
+    this._cardElement = document
+      .querySelector(this._cardSelector)
+      .cloneNode(true).content.firstElementChild;
+    this._likeButton = this._cardElement.querySelector(".card__like-button");
   }
 
   _setEventListeners() {
-    this._likeButton.addEventListener("click", () => this._handleLikeIcon());
-    this._deleteButton.addEventListener("click", () =>
-      this._handleDeleteIcon()
+    this._likeButton.addEventListener("click", () => {
+      this._handleLikeButton();
+    });
+
+    this._cardImageEl.addEventListener("click", () => {
+      this._handleCardClick({ name: this._name, link: this._link });
+    });
+
+    this._cardTrashButton = this._cardElement.querySelector(
+      ".card__trash-button"
     );
-    this._cardImage.addEventListener("click", () => {
-      this._handleImageClick(this);
+    this._cardTrashButton.addEventListener("click", () => {
```

Comment:
Needs correction: I don't see the cards on your page `TypeError: Cannot read properties of null (reading 'addEventListener')
    at Card._setEventListeners (Card.js:37:27)
    at Card.getView (Card.js:80:10)
    at renderCard (index.js:106:17)
    at renderer (index.js:112:22)
    at Array.forEach (<anonymous>)
    at Section.renderItems (Section.js:8:11)
    at index.js:142:17`